drivers/arm/i2c_master.h

```diff
@@ -27,7 +27,7 @@
 #include "ch.h"
 #include <hal.h>
 
-#if defined(STM32F1XX) || defined(STM32F1xx) || defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32L0xx) || defined(STM32L1xx)
+#if defined(STM32F1XX) || defined(STM32F1xx) || defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32F4XX) || defined(STM32L0xx) || defined(STM32L1xx)
```

Comment:
Some quick grepping reveals that these defines should perhaps all be uppercase - eg. `STM32F1xx` doesn't exist in lib/chibios or chibios-contrib, but `STM32F1XX` does.

Reply:
yes, but I'm not sure why it was defined as lower cases before. Chibios does not use lower case dfinitions.

Reply:
It probably simply wasn't caught until now, I guess. There are currently no F2, L0 or L1 boards in QMK, so the fact that those defines here don't exist either, has gone unnoticed.

Reply:
It was copy-pasted from somewhere, though in the original ticket, it doesnt seem like its in the linked blog https://github.com/qmk/qmk_firmware/pull/6262#issuecomment-508912088

Reply:
STM32F1xx does not exist in the CMSIS and ST's HAL library. They should come form stm32arduino library.

Reply:
Changing the rest of the defines is kinda out of scope for this PR. The new addition is consistent with what ChibiOS, so lets tackle any other change separately.